src/System.Memory/src/System/Buffers/ArrayMemoryPool.ArrayMemoryPoolBuffer.cs

```diff
@@ -67,44 +64,39 @@ sealed override bool TryGetArray(out ArraySegment<T> segment)
                 return true;
             }
 
-            public sealed override MemoryHandle Pin(int byteOffset = 0)
+            public sealed override MemoryHandle Pin(int elementIndex = 0)
             {
                 unsafe
                 {
-                    Retain(); // this checks IsDisposed
+                    while (true)
```

Comment:
Remove the reference count here.

Reply:
This is an internal type, the implementation of which we will greatly simplify, outside this PR. I want to get the API surface area changes in first so that they can be consumed up stack.